src/extra/others/fragment/lv_fragment.c

```diff
@@ -0,0 +1,392 @@
+};
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+static manager_stack_t * item_new(const lv_fragment_class_t * cls);
+
+static lv_fragment_t * item_create_fragment(lv_fragment_manager_t * manager, manager_stack_t * item,
+                                            void * args);
+
+static void item_create_obj(lv_fragment_manager_t * manager, manager_stack_t * item, lv_obj_t * parent,
+                            const lv_obj_class_t * check_type);
+
+static void item_destroy_obj(lv_fragment_manager_t * manager, manager_stack_t * item);
+
+static void item_destroy_fragment(manager_stack_t * item);
+
+static void obj_cb_delete(lv_event_t * event);
+
+static void fragment_destroy_obj(lv_fragment_t * fragment);
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+/**********************
+ *      MACROS
+ **********************/
+
+
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+lv_fragment_manager_t * lv_fragment_manager_create(lv_obj_t * container, lv_fragment_t * parent)
+{
+    LV_ASSERT(container);
+    lv_fragment_manager_t * instance = lv_mem_alloc(sizeof(lv_fragment_manager_t));
+    instance->parent = parent;
+    instance->container = container;
+    instance->top = NULL;
+    return instance;
+}
+
+void lv_fragment_manager_del(lv_fragment_manager_t * manager)
+{
+    LV_ASSERT(manager);
+    manager_stack_t * top = manager->top;
+    while(top) {
+        LV_ASSERT(top->cls);
+        item_destroy_obj(manager, top);
+        item_destroy_fragment(top);
+        struct manager_stack_t * prev = top->prev;
+        lv_mem_free(top);
+        top = prev;
+    }
+    lv_mem_free(manager);
+}
+
+void lv_fragment_manager_push(lv_fragment_manager_t * manager, const lv_fragment_class_t * cls, void * args)
+{
+    LV_ASSERT(manager);
+    LV_ASSERT(cls);
+    manager_stack_t * item = item_new(cls);
+    lv_obj_t * parent = manager->container;
+    item_create_fragment(manager, item, args);
+    /* Destroy object of previous screen */
+    if(manager->top) {
+        item_destroy_obj(manager, manager->top);
+    }
+    item_create_obj(manager, item, parent, NULL);
+    manager_stack_t * top = manager->top;
+    item->prev = top;
+    manager->top = item;
+}
+
+void lv_fragment_manager_replace(lv_fragment_manager_t * manager, const lv_fragment_class_t * cls,
+                                 void * args)
+{
+    LV_ASSERT(manager);
+    LV_ASSERT(cls);
+    manager_stack_t * top = item_new(cls);
+    item_create_fragment(manager, top, args);
+    manager_stack_t * old = manager->top;
+    if(old) {
+        item_destroy_obj(manager, old);
+        item_destroy_fragment(old);
+        lv_mem_free(old);
+    }
+    manager->top = top;
+    item_create_obj(manager, top, manager->container, NULL);
+}
+
+size_t lv_fragment_manager_size(lv_fragment_manager_t * manager)
+{
+    size_t size = 0;
+    for(manager_stack_t * cur = manager->top; cur; cur = cur->prev) {
```

Comment:
I fond that in XC16 compiler (Microchip PIC) this declaration doesn't work. 
```suggestion
    manager_stack_t * cur;
    for(cur = manager->top; cur; cur = cur->prev) {
```